service/collector/client.go

```diff
@@ -0,0 +1,47 @@
+package collector
+
+import (
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/credential"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// kubernetesCredentialNamespace is the namespace in which we store credentials.
+	kubernetesCredentialNamespace = "giantswarm"
+
+	// kubernetesLabelSelector is the label selector we use to retrieve credentials.
+	kubernetesLabelSelector = "app=credentiald"
+)
+
+// getUniqueClientSets fetches all unique Azure clients.
+func getUniqueClientSets(k8sClient kubernetes.Interface, environmentName string) ([]*client.AzureClientSet, error) {
```

Comment:
I do not see any check which ensure client sets returned by this function are unique. Would be good to rename this function to `getClientSets` to avoid any confusion.